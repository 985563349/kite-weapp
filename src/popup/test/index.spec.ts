import path from 'node:path';
import simulate from 'miniprogram-simulate';

describe('popup', () => {
  const KPopup = simulate.load(path.resolve(__dirname, '../popup'), 'k-popup', {
    rootPath: path.resolve(__dirname, '../../'),
  });

  test('should render popup add match snapshot', async () => {
    const comp = simulate.render(
      simulate.load({
        usingComponents: {
          'k-popup': KPopup,
        },
        template: `<k-popup visible />`,
      })
    );
    comp.attach(document.createElement('parent-wrapper'));

    await simulate.sleep(16 * 3); // wait three frames for the animation to complete.
    expect(comp.toJSON()).toMatchSnapshot();
  });
});
