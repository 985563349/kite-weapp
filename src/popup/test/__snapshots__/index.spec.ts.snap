// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popup should render popup add match snapshot 1`] = `
<main>
  <wx-root-portal
    enable="{{false}}"
  >
    <wx-view
      ariaLabel="close"
      ariaRole="button"
      class="k-overlay k-fade-enter-to k-fade-enter-active enter-to-class enter-active-class k-class"
      id=""
      style="-webkit-transition-duration:300ms;transition-duration:300ms;z-index:11000"
      bind:tap="onTap"
      catch:touchmove="noop"
      bind:transitionend="onTransitionEnd"
    />
  </wx-root-portal>
  <wx-root-portal
    enable="{{false}}"
  >
    <wx-view
      class="k-popup k-popup--center k-center-enter-to k-center-enter-active enter-to-class enter-active-class k-class"
      id=""
      style="-webkit-transition-duration:300ms;transition-duration:300ms;z-index:11000"
      catch:touchmove="noop"
      bind:transitionend="onTransitionEnd"
    />
  </wx-root-portal>
</main>
`;
